pkg/driver/node.go

```diff
@@ -743,10 +744,15 @@ func (d *nodeService) getVolumesLimit() int64 {
 	if isNitro {
 		enis := d.metadata.GetNumAttachedENIs()
 		nvmeInstanceStoreVolumes := cloud.GetNVMeInstanceStoreVolumesForInstanceType(instanceType)
-		availableAttachments = availableAttachments - enis - blockVolumes - nvmeInstanceStoreVolumes
+		availableAttachments = availableAttachments - enis - nvmeInstanceStoreVolumes
+	}
+
+	if blockVolumes < availableAttachments {
```

Comment:
This seems excessively complicated compared to just doing `if availableAttachments < 0`, is there a reason we're not just doing that?